request-cert/main.go

```diff
@@ -52,53 +51,103 @@ func main() {
 
 	// Check certificate type.
 	var template *x509.CertificateRequest
-	var filePrefix string
+	var filename, csrName string
 	var wantServerAuth bool
 
+	hostname, err := os.Hostname()
```

Comment:
I'd rather address it later and keep the `start with a single replica` limitation in place for now. It doesn't apply to cockroach nodes, just client apps so I think it's ok for a bit. Given that I'm not done with k8s yet, I expect I'll get to it pretty soon.